boards: stm32f746g_disco: Enable Ethernet

chrta opened the pull request:
This adds Ethernet support for the ST STM32F746G Discovery board.

Also a minor typo in eth_stm32_hal.c is fixed and some information are added to the board yaml file.

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

chrta committed:
boards: stm32f746g_disco: Update board yaml

This adds the following information:
- sram size
- flash size
- gpio support

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

chrta committed:
drivers: ethernet: stm32: Fix typo in comment

Fix minor typo in code comment.

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

chrta committed:
boards: stm32f746g_disco: Enable Ethernet

This adds Ethernet support for the ST STM32F746G Discovery board.

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

jukkar reviewed (commented):
Looks ok, note that sanitychecker fails as you have a commit without anything in body.

pfalcon commented on boards/arm/stm32f746g_disco/stm32f746g_disco.yaml:
```diff
@@ -5,3 +5,9 @@ arch: arm
+supported:
+  - netif:eth
+  - gpio
+
```
Probably this extra line worth fixing.

erwango commented on boards/arm/stm32f746g_disco/doc/stm32f746g_disco.rst:
```diff
@@ -100,6 +100,8 @@ The Zephyr stm32f746g_disco board configuration supports the following hardware
 +-----------+------------+-------------------------------------+
 | GPIO      | on-chip    | gpio                                |
 +-----------+------------+-------------------------------------+
+| ETHERNET  | on-chip    | Ethernet                            |
```
Can you update pinmuxing section too?

chrta replied:
Sure. Can you point be to a board doc and the relevant section, so i see what you are expecting?

erwango replied:
cf: https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/nucleo_f429zi/doc/nucleof429zi.rst#default-zephyr-peripheral-mapping

chrta replied:
Should be fixed now

erwango reviewed (changes requested):
One comment otherwise is fine

jukkar reviewed (approved):
LGTM

chrta committed:
boards: stm32f746g_disco: Update peripheral mapping doc

Sync peripheral mapping documentation with the Nucleo F429ZI board doc
and add missing information about Ethernet and the button.

Signed-off-by: Christian Taedcke <hacking@taedcke.com>
